lib/matplotlib/_pylab_helpers.py

```diff
@@ -47,6 +59,50 @@ def get_fig_manager(cls, num):
         return manager
 
     @classmethod
+    def show_all(cls, block=None):
+        """
+        Show all figures.  If *block* is not None, then
+        it is a boolean that overrides all other factors
+        determining whether show blocks by calling mainloop().
+        The other factors are:
+        it does not block if run inside ipython's "%pylab" mode
+        it does not block in interactive mode.
+        """
+
+        managers = cls.get_all_fig_managers()
+        if not managers:
+            return
+
+        for manager in managers:
+            manager.show()
+
+        if block is not None:
+            if block:
+                manager._mainloop()
+            return
+
+        from matplotlib import pyplot
+        try:
+            ipython_pylab = not pyplot.show._needmain
+            # IPython versions >= 0.10 tack the _needmain
+            # attribute onto pyplot.show, and always set
+            # it to False, when in %pylab mode.
+            ipython_pylab = ipython_pylab and get_backend() != 'WebAgg'
+            # TODO: The above is a hack to get the WebAgg backend
+            # working with ipython's `%pylab` mode until proper
+            # integration is implemented.
+        except AttributeError:
+            ipython_pylab = False
+
+        # Leave the following as a separate step in case we
+        # want to control this behavior with an rcParam.
+        if ipython_pylab:
+            block = False
+
+        if not is_interactive() or get_backend() == 'WebAgg':
+            manager._mainloop()
```

Comment:
Most of this method I just copy pasted from the ``ShowBase``.  The mainloop blocks execution, which prevents the script from dying.  If not interactive we don't need it as the code won't die.  Anyway, as I said, just a copy-paste... saying which, I copy-pasted a bit too much, i.e. get_backend doesn't exist in this scope... it should get it from the manager...